feat(slack): Enrich announcement output

aexvir opened the pull request:
It adds:
* Release title, pointing to the repository (and tag if available)
* Bold section titles, with their correspondent emoji
* Section context, summing up the number of commits and referenced issues
* Commit SHA, link and issue references
* Section dividers

Closes #164

fallion commented on internal/providers.go:
```diff
@@ -0,0 +1,34 @@
+
+// GetGitRemote sucks, I hate to do this, but to do it properly it needs a lot of re-architecting
+func GetGitRemote() GitRemote {
+	project, isGitHub := os.LookupEnv("GITHUB_REPOSITORY")
```
Can this be moved into https://github.com/outillage/integrations ?

aexvir replied:
Yep, makes sense, I didn't notice that package

aexvir replied:
Done

fallion commented on internal/providers.go:
```diff
@@ -0,0 +1,34 @@
+package internal
```
This file shouldn't be at the root of internal

fallion replied:
Something like `internal/providers/git_remote.go` would be better.

aexvir replied:
> This file shouldn't be at the root of internal

What's wrong with this? Is some kind of convention?

fallion replied:
Just division `package providers` means that it will be used as `providers.GetGitRemote()` as opposed to `internal.GetGitRemote()`. Providers feels more explicit.

aexvir replied:
Moved to external package

fallion commented on internal/providers.go:
```diff
@@ -0,0 +1,34 @@
+func GetGitRemote() GitRemote {
+	project, isGitHub := os.LookupEnv("GITHUB_REPOSITORY")
+
+	if isGitHub {
```
Is this covered by tests?

fallion commented on internal/text/build_heading.go:
```diff
@@ -3,17 +3,20 @@ package text
 	builder.WriteString(icon)
 
-	heading := fmt.Sprintf("%v", sectionHeadings[category].title)
+	heading := fmt.Sprintf("%v ", internal.PredefinedSections[category].Title)
```
This whitespace causes the expected output tests to fail

fallion commented:
Any update on this?

aexvir commented:
Rebased, fixed a missing closing bracket that was legit making the tests fail, but now I'm not getting it. It's passing locally just fine, but not in DroneCI

```bash
Running tool: /usr/local/bin/go test -timeout 30s github.com/outillage/release-notary/internal/slack -run ^(TestGenerateReleaseNotes)$

ok  	github.com/outillage/release-notary/internal/slack	0.014s
Success: Tests passed.
```
